docs/dev-manuals/adr/0003-merge-category-and-product-service.md

```diff
@@ -0,0 +1,52 @@
+---
+sidebar_position: 3
+---
+
+# Merge Category and Product Service
+
+## Context and Problem Statement
+
+Category and product service are tightly coupled:
+The majority of endpoints of each services needs to either invoke the other service or duplicate the others services data:
+A product variant holds a characteristic value, this needs to be either saved in the product or category service. It is valuable to know if product variants with specific characteristic actually exist, when displaying the product categories in the frontend with each category characteristic. This prevents the user from running into empty queries. We only want to show characteristics to filter by, if product variants of the specific characteristic exists. Therefore the category service needs to know about the current product variants.
+Similarly, displaying a product variant includes its characteristic value, for example: size M for a t-shirt. Each product variant request depends on the characteristic value.
+
+A category can only be deleted if no product points to it. Should a category automatically be deleted if the last product referencing the category is deleted?
+
+## Considered Options
+
+1. Save shared information in product service and let category service request the needed information
+1. Save shared information in product service and let category service request the needed information
```

Comment:
"product service" and "category service" must be swapped in this sentence, right? Because right now, this line is only a duplicate of the preceding line.

Reply:
Thanks for pointing it out :)